### DISCUSSION TITLE
Built-in iTwin  widgets/tools

### DISCUSSION THREAD

**Discussion by ec-dev-1** [iTwin/itwinjs-core#2504] [Q&A] [answered]
Hello there, 

This is a question about the built-in tools and widgets. 
I color-coded the ones I have a question of (see attached image):

1.  Tool Setting (circled in blue):
2.  Section Tools (circled in green):
3. Camera On/Off (circled in red):

Is there a documentation page available that explains in details what they are for?
Also, should we decide to not make them available,  is it possible to hide them?

Lastly, for the "orientation cube" on the top right,  is there an event we can listen to when the users are controlling it?
We are planning on toggling a widget on or off when users are manipulating it.

Any links to documentation or tutorials would be highly appreciated.

![Screenshot 2021-10-14 164646](https://user-images.githubusercontent.com/86491141/137393287-098d74f8-408c-4faf-8974-d8a97fd4bcae.png)

**Comment by NancyMcCallB** [comment, marked as answer]
The Tool Settings is automatically populated by tools when they become active, if there are any settings for the user to configure the tool with. We currently don't offer a way to remove it, since without it the user would miss the options available for tools. You can tear it off from its position and dock it in the left or right panel, though, and collapse that panel to hide it.

The tools you see by default in the tool widget are in the BasicToolWidget. I assume you use Create React App with the iTwin Web Viewer template? If so, you can hide any or all of the tools and add tools of your own to the Tool Widget. [https://www.npmjs.com/package/@itwin/web-viewer-react](url)

We don't have any events to listen for on the Navigation Aid, but you can replace that with your own, as well. The source for the delivered Navigation Aids is here: [https://github.com/imodeljs/imodeljs/tree/master/ui/imodel-components/src/imodel-components-react/navigationaids](url)

A guide to developing your own Navigation Aid is here: [https://github.com/imodeljs/imodeljs/blob/master/docs/learning/ui/framework/NavigationAids.md](url)

@bsteinbk may also have some ideas.

**Reply by bsteinbk** [reply]
@ec-dev-1 - See this [README.md](https://github.com/itwin/viewer/tree/master/packages/modules/web-viewer-react)  and look for the optional prop `defaultUiConfig` which allows you to control the tool buttons shown. The options under `contentManipulationTools` control the tool in top left corner, the options under `navigationTools` control the items in top right corner.

**Reply by ec-dev-1** [reply]
Thank you so much!

**Reply by ec-dev-1** [reply]
@bsteinbk   while I'm successful in disabling the tools I wanted to disable, this presented an unwanted side-effect which affected our code immensely.  Could you check if there's is an error with how I am disabling these settings?

let config = {
    contentManipulationTools: {
        verticalItems: {
             sectionTools: false
      }},
    navigationTools: {
         verticalItems: {
                cameraView: false
      }
    }
}
  return (<>
    <Viewer
      contextId={contextId}
      iModelId={iModelId}
      authConfig={authOptions}
      onIModelConnected={onIModelConnected} //called when iModel connects
      viewportOptions={viewportOptions}     //hides the Scrubber UI
      uiProviders={uiProviders}
      defaultUiConfig={config}
      // theme="dark"
    />

**Reply by aruniverse** [reply]
What do you mean by "this presented an unwanted side-effect which affected our code immensely"?

**Reply by aruniverse** [reply]
https://www.itwinjs.org/sandboxes/ArunGeorge/gh%20discussion%202504

**Reply by ec-dev-1** [reply]
@aruniverse   Thanks for pointing me to the example page.  It appears I am doing things correctly.  
By unwanted side-effects, I meant the view doesn't seem to load correctly.  For example, without altering the default UI configs, my app loads as expected (see opening post). However, once I add this prop and the configuration needed,  the app loads in a very "zoomed in" state in what seems to be a overhead or bird's eye view (see image below).  

For now, I resorted to adding CSS styles to hide the elements needed.  I'll continue to debug and see what could be causing this behavior but I thought I'd share what my experience was just in case there was an error with how I am declaring my configurations.
 
![Screenshot 2021-10-15 132924](https://user-images.githubusercontent.com/86491141/137528880-4bbbcc17-e6d8-427e-9bb9-3389ead1eee7.png)

**Reply by aruniverse** [reply]
That might because of the viewportOptions prop you pass in. Without further seeing your code it will be hard to identify the issue. If you can, please create a sandbox where you can reproduce the issue, or share your code with us (publish it to a public github repo) and we'll try to get to the bottom of it.

**Reply by ec-dev-1** [reply]
The viewportOptions being passed in was added to remove the Scrubber UI:

const viewportOptions: IModelViewportControlOptions = {disableDefaultViewOverlay: true};

I also tried disabling all the extra widget providers being passed in to the uiProviders just to see if it had any effect but I received the same results.
This leads me to believe that the issue might be in the onIModelConnected props.

I'll revisit our code there.  Thank you so much for pointing me in the right direction.
